Extensions/shorten_posts.js

```diff
@@ -188,13 +135,54 @@ XKit.extensions.shorten_posts = new Object({
 			$(this).removeClass("xkit-shorten-posts-shortened");
 			$(this).removeClass("xkit-shorten-posts-shortened-show-tags");
 			$(this).addClass("xkit-shorten-posts-embiggened");
-			$(this).css('height', 'auto');
+
+			if (XKit.page.react) {
+				const $article = $(this).find("article");
+				const $avatar = $(this).find(".xkit-shorten-posts-shortened-avatar");
+
+				$avatar.contents().prependTo($article);
+				$avatar.remove();
+				$article.removeClass("post_wrapper");
+				$(this).find(".post_tags").removeClass("post_tags");
+
+				$(this).find(XKit.extensions.shorten_posts.wrapperCss).css("height", "auto");
+				$(this).css("height", "auto");
+			} else {
+				$(this).css('height', 'auto');
+			}
 		});
 
 	},
 
-	short: function(obj, m_height) {
+	react_short: function(obj, height) {
+		var $obj = $(obj);
+		if ($obj.hasClass("xblacklist_blacklisted_post")) { $obj.removeClass("xkit-shorten-posts-shortened-show-tags"); return; }
+
+		const $article = $obj.find("article");
+
+		const container_css = $article.attr("class");
+		const post_id = $obj.attr("data-id");
+
+		$obj.attr("data-old-height", height);
+		$obj.find(XKit.extensions.shorten_posts.wrapperCss).css("height", XKit.extensions.shorten_posts.preferences.height.value + "px");
+		$article.addClass("post_wrapper");
+
+		const $avatar_wrapper = $(`<div class="xkit-shorten-posts-shortened-avatar ${container_css}"></div>`);
+		$article.find(XKit.extensions.shorten_posts.avatarCss).appendTo($avatar_wrapper);
+		$article.before($avatar_wrapper);
+
+		$obj.addClass("xkit-shorten-posts-shortened");
+
+		$article.append("<div class=\"xkit-shorten-posts-embiggen xkit-shorten-posts-embiggen-for-post-" + post_id + "\" data-post-id=\"" + post_id + "\" data-old-height=\"" + height + "\">This post has been shortened. Click here to show the full post</div>");
```

Comment:
all these escapes and string concatenation is a little rough, could this be a template string instead?